docs/testing-operators.md

```diff
@@ -1,120 +1,404 @@
+```
+my-operator
+operator-marketplace
+operator-courier
+operator-lifecycle-manager
+```
 
-**Note**: no explicit number of points or percentage is necessary to achieve before merging _yet_. These are suggestions to improve your operator.
+### Tools
 
 #### operator-courier
 
-The [`operator-courier verify`][courier] command verifies that a set of files is valid and can be bundled and pushed to [quay.io][quay]. Read the [docs][courier-docs] for more information.
+`operator-courier` is used for metadata syntax checking and validation. This can be installed directly from `pip`:
+
+```
+pip3 install operator-courier
+```
+
+#### Quay Login
+
+In order to test the Operator installation flow, store your Operator bundle on [quay.io](https://quay.io). You can easily create an account and use the free tier (public repositories only). To upload your Operator to quay.io a token is needed. This only needs to be done once and can be saved locally. The `operator-courier` repository has a script to retrieve the token:
+
+```
+./operator-courier/scripts/get-quay-token
+
+Username: johndoe
+Password: 
+{"token": "basic abcdefghijkl=="}
+```
+
+A token takes the following form and should be saved in an environment variable:
+
+```
+export QUAY_TOKEN="basic abcdefghijkl=="
+```
+### Linting
+
+`operator-courier` will verify the fields included in the Operator metadata (CSV). The fields can also be manually reviewed according to [the operator CSV documentation](https://github.com/operator-framework/community-operators/blob/master/docs/required-fields.md).
+
+The following command will run `operator-courier` against the bundle directory `my-operator/` from the example above.
+
+```
+operator-courier verify --ui_validate_io my-operator/
+```
+
+If there is no output, the bundle passed `operator-courier` validation. If there are errors, your bundle will not work. If there are warning we still encourage you to fix them before proceeding to the next step.
+
+### Push to quay.io
+
+The Operator metadata in it's bundle format will be uploaded into your namespace in [quay.io](http://quay.io).
+
+> The value for `PACKAGE_NAME` *must* be the same as in the operator's `*package.yaml` file and the operator bundle directory name. Assuming it is `my-operator`, this can be found by running `cat my-operator/*.package.yaml`.
 
-### Upstream operators
+> The `PACKAGE_VERSION` is entirely up for you to decide. Best practice is it coincides with your Operator version.
 
-Operators submitted to the `upstream-community-operators/` directory are tested against a [`minikube`][minikube] instance deployed on a [Travis CI][travis-ci] environment. The OLM is installed locally in this case.
+```
+OPERATOR_DIR=my-operator/
+QUAY_USERNAME=johndoe
```

Comment:
nit: This is only the case for pushing to a single user's working namespace. If any other type of account is used, this should be the namespace of that account. I understand the purpose of simplicity here, but it might be important to call that out.

Reply:
The instruction explicitly ask to upload to user namespace, not to an organization. Unless it's possible to have no personal namespace on quay but push access to an organization it's consistent.

Reply:
Consider that someone most likely will come to this doc when they have a question about some specific issue they are encountering, rather than reading it end to end. In that case, calling this field "QUAY_USERNAME" and not "NAMESPACE" could be confusing. I think just making it more explicit why we are using a username for the namespace would be appropriate, or at least mentioning that this username maps to the quay app namespace.